codersdk/deployment.go

```diff
@@ -154,6 +166,7 @@ type DeploymentValues struct {
 	CacheDir                        serpent.String                       `json:"cache_directory,omitempty" typescript:",notnull"`
 	InMemoryDatabase                serpent.Bool                         `json:"in_memory_database,omitempty" typescript:",notnull"`
 	PostgresURL                     serpent.String                       `json:"pg_connection_url,omitempty" typescript:",notnull"`
+	PostgresAuth                    string                               `json:"pg_auth,omitempty" typescript:",notnull"`
```

Comment:
Just a missing password field, maybe an AWS domain in the URL, but I don't think there's anything that's enough to make an inference.

Reply:
One thing that might be possible is if we detected the URL scheme and had people set `awsiamrds://...` instead of `postgres://`, or add a query param? It is unforunate we need a new flag just for aws auth, but it's probable we would exapand this in the future. My only worry is that just detecting it in the connection string would be non-obvious, since I haven't really seen it anywhere else, but to be fair I haven't really done much research.

Reply:
Yeah I wouldn't wanna change the protocol because that kinda implies it's not speaking `postgres` protocol. I think given we may have other auth schemes in the future we need to support, a flag seems fine to me and better than any parsing options I can think of.